benchmarks/osb/extensions/runners.py

```diff
@@ -0,0 +1,118 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+from .util import parse_int_parameter, parse_string_parameter
+import time
+
+
+def register(registry):
+    registry.register_runner(
+        "custom-vector-bulk", BulkVectorsFromDataSetRunner(), async_runner=True
+    )
+    registry.register_runner(
+        "custom-refresh", CustomRefreshRunner(), async_runner=True
+    )
+    registry.register_runner(
+        "train-model", TrainModelRunner(), async_runner=True
+    )
+    registry.register_runner(
+        "delete-model", DeleteModelRunner(), async_runner=True
+    )
+
+
+class BulkVectorsFromDataSetRunner:
+
+    async def __call__(self, opensearch, params):
+        size = parse_int_parameter("size", params)
+        retries = parse_int_parameter("retries", params, 0) + 1
+
+        for _ in range(retries):
+            try:
+                await opensearch.bulk(
+                    body=params["body"],
+                    timeout='5m'
+                )
+
+                return size, "docs"
+            except:
+                pass
+
+        raise TimeoutError("Failed to submit bulk request in specified number "
+                           "of retries: {}".format(retries))
+
+    def __repr__(self, *args, **kwargs):
+        return "custom-vector-bulk"
+
+
+class CustomRefreshRunner:
+
+    async def __call__(self, opensearch, params):
+        retries = parse_int_parameter("retries", params, 0) + 1
+
+        for _ in range(retries):
+            try:
+                await opensearch.indices.refresh(
+                    index=parse_string_parameter("index", params)
+                )
+
+                return
+            except:
+                pass
```

Comment:
shall we capture the error and log. Also, we should know whether is it transient error or actual error. We should only retry transient error.

Reply:
Sure, I will catch connection timeout error and log a message with it.